src/View3d.ts

```diff
@@ -229,8 +230,9 @@ export class View3d {
     this.image?.onChannelAdded(newChannelIndex);
   }
 
-  setTime(volume: Volume, time: number): void {
-    volume.updateRequiredData({ time: Math.max(0, Math.min(time, volume.imageInfo.times)) });
+  setTime(volume: Volume, time: number, onChannelLoaded?: PerChannelCallback): void {
+    const timeClamped = Math.max(0, Math.min(time, volume.imageInfo.times - 1));
```

Comment:
callers should also be responsible for passing correct times in; thinking about playback and looping.

Reply:
Should this function throw an error or return some error value rather than quietly fixing out-of-bounds time values for the caller?

Reply:
I think it's ok if the responsibility is shared.  This step you have here seems ok to prevent catastrophic problems.  (Tradeoff: if you handle error gracefully, sometimes it hides that there was something actually wrong.  If you raise an exception, you clearly know something is wrong but you crash the app.)